### PR TITLE
Adds chart for incoming proto msgs in new cloud protocol

### PR DESCRIPTION
**Pull request by underhood**
##### Summary
Adds charts showing incoming cloud messages for new cloud protocol.

##### Component Name
ACLK
##### Test Plan
Connect agent to cloud with new protocol used (check with `api/v1/aclk` check field `used-cloud-protocol == New`), in such case chart should show what messages are incoming from cloud.

### CODE REVIEW

**Comment by MrZammler** [comment]
Hi Timo! I'm checking, I think it runs fine:

image

(Not sure btw of the `cmd` messages, but I'm on staging if that counts)

**Comment by underhood** [comment]
@MrZammler the `cmd` message is the only old cloud message that is still used in new cloud protocol. It allows access to the `http` endpoints trough the old `cmd` message. That is then broken into finer detail by charts above (there is a chart saying which http endpoint was queried)

**Comment by MrZammler** [comment]
Thanks! Btw, will we also add outgoing messages?

**Comment by underhood** [comment]
Yes I intend to do that in next PRs that will come

File: aclk/aclk_stats.c [hunk 1]
```diff
@@ -245,8 +253,57 @@ static void aclk_stats_query_time(struct aclk_metrics_per_sample *per_sample)
...
+#ifdef ENABLE_NEW_CLOUD_PROTOCOL
+const char *rx_handler_get_name(size_t i);
+static void aclk_stats_newproto_rx(uint32_t *aclk_proto_tx_msgs_sample)
+{
+    static RRDSET *st = NULL;
+
+    if (unlikely(!st)) {
+        st = rrdset_create_localhost(
+            "netdata", "aclk_protobuf_tx_types", NULL, "aclk", NULL, "Received new cloud architecture messages by their type.", "msg/s",
+            "netdata", "stats", 200010, localhost->rrd_update_every, RRDSET_TYPE_LINE);
```
Comment (ilyam8) [inline aclk/aclk_stats.c:265]:
```suggestion
            "netdata", "stats", 200010, localhost->rrd_update_every, RRDSET_TYPE_STACKED);
```

Perhaps stacked? So we can see the total number of msg/s (if that makes sense).

File: aclk/aclk_stats.c [hunk 2]
```diff
@@ -245,8 +253,57 @@ static void aclk_stats_query_time(struct aclk_metrics_per_sample *per_sample)
...
 
+#ifdef ENABLE_NEW_CLOUD_PROTOCOL
+const char *rx_handler_get_name(size_t i);
+static void aclk_stats_newproto_rx(uint32_t *aclk_proto_tx_msgs_sample)
+{
+    static RRDSET *st = NULL;
+
+    if (unlikely(!st)) {
+        st = rrdset_create_localhost(
+            "netdata", "aclk_protobuf_tx_types", NULL, "aclk", NULL, "Received new cloud architecture messages by their type.", "msg/s",
```
Comment (ilyam8) [inline aclk/aclk_stats.c:264]: > msg/s

Is this really per second? I see that the dim algorithm is absolute. Do you calculate delta internally?

File: aclk/aclk.c [hunk 3]
```diff
@@ -763,8 +763,9 @@ void *aclk_main(void *ptr)
         return NULL;
     }
 
+    unsigned int proto_hdl_cnt;
```
Comment (MrZammler) [inline aclk/aclk.c:766]: Hi Timo! Should this be initialized? Or inside the ifdef below? It is used later on in line 808... ?

File: aclk/aclk_stats.c (hunk 2 above)
Comment (underhood) [reply aclk/aclk_stats.c:264]: the divisor used is `localhost->rrd_update_every`

File: aclk/aclk_stats.c (hunk 1 above)
Comment (underhood) [reply aclk/aclk_stats.c:265]: i was thinking about that actually not sure why i choose line in the end. I will change it to stacked

File: aclk/aclk.c (hunk 3 above)
Comment (underhood) [reply aclk/aclk.c:766]: in such case it is passed but unused
```
void aclk_stats_thread_prepare(int query_thread_count, unsigned int proto_hdl_cnt)
{
#ifndef ENABLE_NEW_CLOUD_PROTOCOL
    UNUSED(proto_hdl_cnt);
#endif
```

**Comment by underhood** [comment]
rebased on latest master hoping to fix stuck CI (which gets stuck even after rerun all jobs)

**Review by MrZammler — APPROVED** [review]

**Review by odynik — APPROVED** [review]
### Netdatacli aclk-state

image

### Tested on staging environment

image

**Pull request merged**